### PR TITLE
Feature/add client lib functions

### PR DESCRIPTION
**Pull request by surabhil**
- Add following functions to the verifier module:
  hashParams - to hash the parameters sent to the verifier
  getAccessTokenFromVerifier 

- Add updateUsageLog function to call the verifier usage handle to update the usage for an instrument on ORE blockchain

### CODE REVIEW

**Review by blackwatertepes — CHANGES_REQUESTED** [review]
signVoucher needs to be moved out of apimarket section of the library, since it's about to be removed.

File: src/verifier.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const {
+        endpoint,
+        oreAccessToken,
+        method,
+        additionalParameters,
+        accessTokenTimeout
+    } = await result.json();
+
+
+    if (!oreAccessToken || oreAccessToken === undefined) {
```
Comment (blackwatertepes) [inline src/verifier.js]: this can be simpler `if (!oreAccessToken) {`

File: src/verifier.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+//Call the Verifier to verify the client request and return an ore-access-token to access a particular right
+async function getAccessTokenFromVerifier(verifierEndpoint, instrument, right, hashedParams) {
+    let errorMessage;
+    let result;
+    let signature;
+
+    try {
+        signature = await this.signVoucher(instrument.id);
+    } catch (error) {
+        throw new Error(`${error.message}`);
```
Comment (blackwatertepes) [inline src/verifier.js]: This isn't doing anything extra. You can just remove the entire try/catch block, and accomplish the same thing.

**Review by blackwatertepes — CHANGES_REQUESTED** [review]
We also need to remove all references to "voucher". The OreJS library should be only protocol specific, and not care about Offers/Vouchers.

**Review by traylewin — APPROVED** [review]

**Review by blackwatertepes — APPROVED** [review]
👍

**Pull request merged**